packages/xo-web/src/common/xo/index.js

```diff
@@ -968,11 +972,49 @@ export const deleteTemplates = templates =>
   confirm({
     title: _('templateDeleteModalTitle', { templates: templates.length }),
     body: _('templateDeleteModalBody', { templates: templates.length }),
-  }).then(
-    () =>
-      Promise.all(map(resolveIds(templates), id => _call('vm.delete', { id }))),
-    noop
-  )
+  }).then(async () => {
+    const defaultTemplates = []
+    let nErrors = 0
+    await Promise.all(
+      map(resolveIds(templates), id =>
+        _call('vm.delete', { id }).catch(reason => {
+          if (vmIsTemplate.is(reason)) {
+            defaultTemplates.push(id)
+          } else {
+            nErrors++
+          }
+        })
+      )
+    )
+
+    const nDefaultTemplate = defaultTemplates.length
+    if (nDefaultTemplate === 0 && nErrors === 0) {
+      return
+    }
+
+    const showError = () =>
+      error(
+        _('failedToDeleteTemplatesTitle', { nTemplates: nErrors }),
+        _('failedToDeleteTemplatesMessage', { nTemplates: nErrors })
+      )
+
+    return nDefaultTemplate === 0
+      ? showError()
+      : confirm({
+        title: _('DeleteDefaultTemplatesTitle', { nDefaultTemplate }),
+        body: _('DeleteDefaultTemplatesMessage', { nDefaultTemplate }),
+      }).then(() => {
+        if (nErrors !== 0) {
+          showError()
+        }
```

Comment:
What if force deleting the default templates triggers genuine errors? Don't you want to show it in the error notification?